eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/views/FilterStateItemView.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.filter.views;
+
+import com.schedjoules.eventdiscovery.databinding.SchedjoulesViewFilterItemBinding;
+import com.schedjoules.eventdiscovery.framework.filter.categoryoption.CategoryOption;
+import com.schedjoules.eventdiscovery.framework.filter.filterstate.FilterState;
+import com.schedjoules.eventdiscovery.framework.utils.smartview.SmartView;
+import com.schedjoules.eventdiscovery.framework.widgets.Highlightable;
+
+
+/**
+ * Represents the View for a category filter state item.
```

Comment:
I think this is not a general view for an item view, it is specifically for the "All" item because of the inverse logic of highlighting in it, that's why I thought in the first place that it can be worth creating, to contain that. Thinking about inverse logic, '!` and `true` and 'false` in an already complex class can be error prone imo. So that was the motivation.
So this could be something like `AllSelectionItemView`.